lisa/tools/ethtool.py

```diff
@@ -258,46 +258,62 @@ def _get_device_list(self) -> None:
             if not netdir:
                 continue
             cmd_result = self.node.execute(f"ls {netdir}")
-            if cmd_result.exit_code != 0:
-                raise LisaException(
-                    "Could not find the network device under path {netdir}"
-                    f" exit_code: {cmd_result.exit_code}"
-                    f" stderr: {cmd_result.stderr}"
-                )
-            self._device_list.append(cmd_result.stdout)
+            cmd_result.assert_exit_code(
+                message=f"Could not find the network device under path {netdir}"
+            )
 
-        if not self._device_list:
+            self._device_set.add(cmd_result.stdout)
+
+        if not self._device_set:
             raise LisaException("Did not find any synthetic network interface.")
 
+        return self._device_set
+
     def get_device_channels_info(
-        self,
-        interface: str,
+        self, interface: str, force: bool = False
     ) -> DeviceChannel:
-        if interface in self._device_channel_map.keys():
+        if (not force) and (interface in self._device_channel_map.keys()):
             return self._device_channel_map[interface]
 
-        result = self.run(f"-l {interface}")
-        if result.exit_code != 0:
-            raise LisaException(
-                f"ethtool -l {interface} command got non-zero"
-                f" exit_code: {result.exit_code}"
-                f" stderr: {result.stderr}"
-            )
+        result = self.run(f"-l {interface}", force_run=force)
+        if (result.exit_code != 0) and ("Operation not supported" in result.stderr):
+            raise SkippedException
+        result.assert_exit_code(
+            message=f"Couldn't get device {interface} channels info."
+        )
+
+        if interface in self._device_channel_map.keys():
```

Comment:
The two lines are not necessary, in next lines, the value will be replaced.

Reply:
A new object is created in next line before inserting the value in list. So Need to delete the existing object.

Reply:
The item in dict will be replaced each time. `a[b] = "c"`

Reply:
> The item in dict will be replaced each time. `a[b] = "c"`

Concern is not about replacing. I am creating a new object for this value "C". So I just want to delete the object that was in a[b]. Does that make sense?

Reply:
It doesn't need, the `del` keyword is to delete key in the dict, not to delete the object. And Python manage memory itself, it doesn't need to delete an object, unless it uses other resource than memory.